code/node_modules/abc_doc.js

```diff
@@ -0,0 +1,35 @@
+class ABCDoc {
+
+    constructor(source_directory, target_file) {
+        if (this.constructor === ABCDoc) {
+            throw new TypeError('Abstract class "ABCDoc" cannot be instantiated directly.'); 
+        }
+        this.target_file = target_file
+        this.source_dir = source_directory
+        this.doc = this.get_doc()
+        this.write().then(_ => {
+            this.check_written()
+        }).catch(err => {env.logger.debug(err)})
+    }
```

Comment:
The class constructor (which is run when the object is new) is almost agnostic to it's downstream components, but it requires a source_directory and target_file, which it should recieve via super(source_directory, target_file)

`write()` and `check_written()` are both agnostic of their downstream components, and can be independently defined here.